setup.py

```diff
@@ -1,186 +1,94 @@
-    # no setuptools installed
-    from distutils.core import setup, Command
-    _have_setuptools = False
+        mod = importlib.import_module(key)
+        ver = LooseVersion(mod.__version__)
+        if ver < req_ver:
+            raise RuntimeError(REQ_NOT_MET_MSG.format(key, ver, req_ver))
+    except ImportError:
+        pass
+    except AttributeError:
+        raise RuntimeError(REQ_NOT_MET_MSG.format(key, ver, req_ver))
+
+INSTALL_REQUIREMENTS = SETUP_REQUIREMENTS.copy()
+INSTALL_REQUIREMENTS.update({'pandas': '0.19',  # released October 2016
+                             'patsy': '0.4.0',  # released July 2015
+                             })
+
+CYTHON_MIN_VER = '0.24'  # released Apr 2016
 
-if _have_setuptools:
-    setuptools_kwargs = {"zip_safe": False}
-else:
-    setuptools_kwargs = {}
-    if sys.version_info[0] >= 3:
-        sys.exit("Need setuptools to install statsmodels for Python 3.x")
+SETUP_REQUIRES = [k + '>=' + v for k, v in SETUP_REQUIREMENTS.items()]
+INSTALL_REQUIRES = [k + '>=' + v for k, v in INSTALL_REQUIREMENTS.items()]
 
+EXTRAS_REQUIRE = {'build': ['cython>=' + CYTHON_MIN_VER],
+                  'develop': ['cython>=' + CYTHON_MIN_VER],
+                  'docs': ['sphinx', 'nbconvert', 'jupyter_client',
+                           'ipykernel', 'matplotlib', 'nbformat', 'numpydoc',
+                           'pandas-datareader']}
 
-curdir = os.path.abspath(os.path.dirname(__file__))
-README = open(pjoin(curdir, "README.rst")).read()
-CYTHON_EXCLUSION_FILE = 'cythonize_exclusions.dat'
+###############################################################################
+# Values that rarely change
+###############################################################################
 
 DISTNAME = 'statsmodels'
 DESCRIPTION = 'Statistical computations and models for Python'
+with open(pjoin(os.getcwd(), 'README.rst')) as readme:
+    README = readme.read()
 LONG_DESCRIPTION = README
 MAINTAINER = 'Josef Perktold, Chad Fulton, Kerby Shedden'
 MAINTAINER_EMAIL = 'pystatsmodels@googlegroups.com'
 URL = 'https://www.statsmodels.org/'
 LICENSE = 'BSD License'
 DOWNLOAD_URL = ''
 
-# These imports need to be here; setuptools needs to be imported first.
-from distutils.extension import Extension
-from distutils.command.build import build
-from distutils.command.build_ext import build_ext as _build_ext
-
-
-class build_ext(_build_ext):
-    def build_extensions(self):
-        numpy_incl = pkg_resources.resource_filename('numpy', 'core/include')
-
-        for ext in self.extensions:
-            if (hasattr(ext, 'include_dirs') and
-                    not numpy_incl in ext.include_dirs):
-                ext.include_dirs.append(numpy_incl)
-        _build_ext.build_extensions(self)
-
-
-def generate_cython():
-    cwd = os.path.abspath(os.path.dirname(__file__))
-    print("Cythonizing sources")
-    p = subprocess.call([sys.executable,
-                          os.path.join(cwd, 'tools', 'cythonize.py'),
```

Comment:
I think this means we can get rid of tools.cythonize

Reply:
Yes, in due course